### PR TITLE
*: strip the RefCell off the LocalReader in ServerRaftStoreRouter

### LINKED ISSUE #13546: It is better to remove RefCell of LocalReader in ServerRaftStoreRouter
## Development Task
LocalReader is wrapped by a RefCell which enables it to be called by immutable reference. But it adds some extra abstraction and it is unnecessarily callbed by immutable reference.

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #13546

What's Changed:

The current old use RefCell to enable LocalReader be called immutably.
This PR remove it as it adds extra abstractions and RefCell is not pretty safe.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/storage/txn/commands/prewrite.rs [hunk 1]
```diff
@@ -35,7 +35,6 @@ use crate::storage::{
     types::PrewriteResult,
     Context, Error as StorageError, ProcessResult, Snapshot,
 };
-
```
Comment (BusyJay) [inline src/storage/txn/commands/prewrite.rs:38]: Why remove this?

**Review by BusyJay — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**